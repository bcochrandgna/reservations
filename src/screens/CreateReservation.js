// @flow
import React, { PureComponent } from 'react';
import { SafeAreaView } from 'react-native';
import { Mutation } from 'react-apollo';
import { Icon } from 'react-native-elements';
import styles from '../styles/screens/CreateReservation';
import CREATE_RESERVATION from '../data/mutations/createReservation';
import i18n from '../helpers/i18n';
import { Reservation } from '../types/Reservation';
import ReservationForm from '../components/ReservationForm';
import CreateReservationHeader from '../components/CreateReservationHeader';

type Props = {
  navigation: any,
};
type TabBarIconArguments = {
  tintColor: string,
};

class CreateReservation extends PureComponent<Props> {
  static navigationOptions = {
    tabBarIcon: ({ tintColor }: TabBarIconArguments) => (
      <Icon name="playlist-add" color={tintColor} />
    ),
    title: i18n.t('create_reservation'),
  };

  handleCreateReservation = (createReservation: (reservation: Reservation) => any) => async (
    reservation: Reservation,
  ) => {
    const { navigation } = this.props;
    await createReservation({
      variables: {
        name: reservation.name,
        hotelName: reservation.hotelName,
        arrivalDate: reservation.arrivalDate,
        departureDate: reservation.departureDate,
      },
    });

    navigation.goBack();
  };

  render() {
    return (
      <SafeAreaView style={styles.container}>
        <CreateReservationHeader />
        <Mutation mutation={CREATE_RESERVATION}>
          {createReservation => (
            <ReservationForm onUpdate={this.handleCreateReservation(createReservation)} />
          )}
        </Mutation>
      </SafeAreaView>
    );
  }
}

export default CreateReservation;
