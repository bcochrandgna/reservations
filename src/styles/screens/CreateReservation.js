import { StyleSheet } from 'react-native';
import { screenBackgroundColor } from '..';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: screenBackgroundColor,
  },
});
