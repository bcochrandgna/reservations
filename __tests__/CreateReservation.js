import React from 'react';
import { shallow } from 'enzyme';
import CreateReservation from '../src/screens/CreateReservation';

describe('CreateReservation', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CreateReservation />);
    expect(wrapper).toMatchSnapshot();
  });
});
