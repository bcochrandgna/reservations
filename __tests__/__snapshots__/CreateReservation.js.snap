// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateReservation renders correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <CreateReservationHeader />
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "CREATE_RESERVATION",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "data",
                      },
                      "value": Object {
                        "fields": Array [
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                            },
                          },
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "hotelName",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "hotelName",
                              },
                            },
                          },
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "arrivalDate",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "arrivalDate",
                              },
                            },
                          },
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "departureDate",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "departureDate",
                              },
                            },
                          },
                        ],
                        "kind": "ObjectValue",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createReservation",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "hotelName",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "arrivalDate",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "departureDate",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "hotelName",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "arrivalDate",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "departureDate",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 394,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Mutation>
</SafeAreaView>
`;
